node/resolvers/search/index.ts

```diff
@@ -472,25 +476,47 @@ export const queries = {
     if (!args.identifier) {
       throw new UserInputError('No product identifier provided')
     }
+    const salesChannel = rawArgs.salesChannel
+
+    let vtexSegment: string | undefined = ""
+
+    let cookie: SegmentData | undefined = ctx.vtex.segment
 
     const { field, value } = args.identifier
     let products = [] as SearchProduct[]
+    if (!cookie || (!cookie?.regionId && rawArgs.regionId)) {
+      cookie = {
+        regionId: rawArgs.regionId,
+        // @ts-ignore
```

Comment:
I ignore the ts compilation here because the segment object uses channel as a number but in multiple cases, it's handled as a string, so I prefer to use as a string to guarantee that this object encoded is going to behave the same as vtex segment